vowpalwabbit/core/src/reductions/cb/details/large_action/compute_dot_prod_avx2.cc

```diff
@@ -0,0 +1,176 @@
+  __m256i indices2 = _mm256_add_epi64(feature_indices2, offsets);
+
+  indices1 = _mm256_add_epi64(_mm256_and_si256(indices1, weights_masks), column_indices);
+  __m256i popcounts1 = popcount64(indices1);
+  indices2 = _mm256_add_epi64(_mm256_and_si256(indices2, weights_masks), column_indices);
+  __m256i popcounts2 = popcount64(indices2);
+
+  // popcounts always fit into 32 bits, so truncate and pack all 16 popcounts.
+  __m256i popcounts = pack64to32(popcounts1, popcounts2);
+  __m256i sparsity_indices = _mm256_slli_epi32(_mm256_and_si256(popcounts, all_ones), 1);
+
+  indices1 = _mm256_add_epi64(indices1, seeds);
+  popcounts1 = popcount64(indices1);
+  indices2 = _mm256_add_epi64(indices2, seeds);
+  popcounts2 = popcount64(indices2);
+
+  popcounts = pack64to32(popcounts1, popcounts2);
+  __m256i value_indices = _mm256_add_epi32(sparsity_indices, _mm256_and_si256(popcounts, all_ones));
+
+  __m256 tmp = _mm256_permutevar8x32_ps(value_maps, value_indices);
+  sums = _mm256_fmadd_ps(feature_values, tmp, sums);
+}
+
+// A data parallel implementation of the foreach_feature that processes 8 features at once.
+float compute_dot_prod_avx2(uint64_t column_index, VW::workspace* _all, uint64_t seed, VW::example* ex)
+{
+  float sum = 0.f;
+  const uint64_t offset = ex->ft_offset;
+  const uint64_t weights_mask = _all->weights.mask();
+
+  __m256 sums = _mm256_setzero_ps();
+  const __m256i column_indices = _mm256_set1_epi64x(column_index);
+  const __m256i seeds = _mm256_set1_epi64x(seed);
+  const __m256i weights_masks = _mm256_set1_epi64x(weights_mask);
+  const __m256i offsets = _mm256_set1_epi64x(offset);
+
+  for (const auto& features : *ex)
+  {
+    const size_t num_features = features.size();
+    size_t i = 0;
+    for (; i + 8 <= num_features; i += 8)
+    {
+      // Unroll the 64-bit indices twice to align with 32-bit values.
+      __m256i indices1 = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(&features.indices[i]));
+      __m256i indices2 = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(&features.indices[i + 4]));
+      // If indices fit into 32 bits, convert indices to 32-bit here can speed up further.
+
+      __m256 values = _mm256_loadu_ps(&features.values[i]);
+      compute8(values, indices1, indices2, offsets, weights_masks, column_indices, seeds, sums);
+    }
+    for (; i < num_features; ++i)
+    {
+      // Handle tail of the loop using scalar implementation.
+      compute1(features.values[i], features.indices[i], offset, weights_mask, column_index, seed, sum);
+    }
+  }
+  const auto& red_features = ex->ex_reduction_features.template get<VW::large_action_space::las_reduction_features>();
+  const auto& interactions =
+      red_features.generated_interactions ? *red_features.generated_interactions : *ex->interactions;
+  for (const auto& ns : interactions)
+  {
+    // TODO: other interactions, ignores & extent_interactions. The following only handles quadratics.
```

Comment:
should we emit some sort of warning if users select LAS + simd + ignore/extend/cubics/other instead of silently ignoring?